shell/platform/android/test/io/flutter/embedding/engine/FlutterEngineTest.java

```diff
@@ -0,0 +1,34 @@
+package test.io.flutter.embedding.engine;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.FlutterJNI;
+import io.flutter.embedding.engine.loader.FlutterLoader;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@Config(manifest=Config.NONE)
+@RunWith(RobolectricTestRunner.class)
+public class FlutterEngineTest {
+  @Test
+  public void itDoesNotCrashIfGeneratedPluginRegistrantIsUnavailable() {
```

Comment:
Should we also have a test when the registrant is available and verify that `registerWith` is called?

Reply:
Would that be a test in the engine or framework?

Reply:
Ideally, both. If you add one here, you will get an extra signal earlier.

Reply:
Spoke offline. Testing infra appears to be too limited to test both the positive and negative case, so we'll leave as-is.